### PR TITLE
Adding  Pre-Commit Hooks (CI)

### PR DESCRIPTION
**Pull request by arjun-kg**
This PR adds pre-commit hooks. There are a ton of changes, but for now they're all trivial like spelling changes, reordering and space. The aim is to bring the codebase to a formatting standard with zero effect to functionality. I'm also using gymnasium's pre-commit as a template. 

Since there are an overwhelming amount of changes, I'll try to do this in phases, and maybe remove hooks which cause too many problems. 

- [x] codespell - Easy for comments but unclear for variable names. I ignore ones that could be interpreted as truncated words.  There are also many French (?) comments. Might do the obvious ones, and then deal with the others in a separate PR.
- [x] flake8 - These seem straightforward, I'll fix soon.
- [x] shebangs - I'm not clear about this. There are executable .bat files without shebangs, not sure on what shebangs to add. I'll read up about this.
- [ ] pydocstyle - Adding docstrings to public functions. 
- [ ] tests - To ensure everything still functions the same.

### CODE REVIEW

**Comment by mwydmuch** [comment]
Hi @arjun-kg, thank you for starting to work on this! Just one remark for the feature, please ensure that `src/vizdoom` directory is excluded from any advance auto-formatting (removing trailing spaces is fine). The thing is that this is mostly an old code, we are not changing it, just fixing compatibility from time to time, and it helps a lot to be able to diff it with some other forks of zdoom. And it is much easier when this code stays the same :)

**Comment by arjun-kg** [comment]
@mwydmuch Got it! I've rolled back some changes, not touching `src` entirely for now, it was too much to manage even if the changes were not significant.

File: pyproject.toml [hunk 1]
```diff
@@ -1,2 +1,31 @@
...
+typeCheckingMode = "basic"
+pythonVersion = "3.7"
+pythonPlatform = "All"
+typeshedPath = "typeshed"
+enableTypeIgnoreComments = true
+
+# This is required as the CI pre-commit does not download the module
+#   Therefore, we have to ignore missing imports
+reportMissingImports = "none"
+reportGeneralTypeIssues = "none"
```
Comment (pseudo-rnd-thoughts) [inline pyproject.toml:29]: I should `"warning"` as it means that we are currently ignoring general type issues

**Comment by pseudo-rnd-thoughts** [comment]
@mwydmuch Im not sure if you have merge permissions still, if not, just approve the PR and poke me on discord and I will merge asap

**Comment by mwydmuch** [comment]
@pseudo-rnd-thoughts Yes, I still have merge permission. @arjun-kg I'm starting to review.

**Review by mwydmuch — APPROVED** [review]
Everything looks good to me

**Pull request merged**